### PR TITLE
feature(API): add transactionAnchorInfo to health

### PR DESCRIPTION
**Pull request by krobi64**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

References #832 

Add an `IPFSDirectoryHashDAO` to the API module and tie in the report to the `GET /health` endpoint

### CODE REVIEW

File: src/API/HealthController.ts [hunk 1]
```diff
@@ -10,10 +13,16 @@ interface HealthObject {
   readonly networkInfo: object
   readonly estimatedSmartFeeInfo: object
   readonly ipfsRetryInfo: object
+  readonly transactionAnchorRetryInfo: AnchorRetryReport | Error
+}
+
+interface Error {
```
[the remark this answers is not in the document]
Comment (krobi64) [reply src/API/HealthController.ts]: that works

File: src/API/API.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { ExchangeConfiguration } from './ExchangeConfiguration'
 import { FileController, FileControllerConfiguration } from './FileController'
 import * as FileDAO from './FileDAO'
 import { HealthController } from './HealthController'
+import * as IPFSDirectoryHashDAO from './IPFSDirectoryHashDAO'
```
Comment (lautarodragan) [inline src/API/API.ts]: can we avoid namespacing the import here?

Comment (krobi64) [reply src/API/API.ts]: I was following Kenny's example. I think since the Symbols are no longer required, I can go back to a normal import

Comment (kennylavender) [reply src/API/API.ts]: I was just experimenting, sorry, didn't make for it to be taken as an example of how we are changing the code.

Comment (krobi64) [reply src/API/API.ts]: Should experiments make it into production code? They become examples of how to do things.

**Comment by krobi64** [comment]
Is #874 a blocker? I knew this would need to be resolved so that we can truly separate the modules. I was already thinking about using RMQ to resolve this. There will be a similar DAO in the Health module, with the difference being that it inserts and updates entries, whereas this DAO only reads.

Refactoring this is a larger task.

@geoffturk what is the priority?

File: src/API/API.ts (hunk 2 above)
Comment (kennylavender) [reply src/API/API.ts]: No, it was a mistake.

**Comment by kennylavender** [comment]
@krobi64 I am good with this being merged before #874 if @lautarodragan is.

I think it might be helpful to move the DAO to the Health module. API can import the DAO from Health, that should give us an easy way to see that we have broken the Module boundaries.

File: src/API/IPFSDirectoryHashDAO.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  readonly getAnchorRetryHealth: getAnchorRetryHealth = async () => {
+    const cursorArray = await this.collection.aggregate([
+      { $match: { attempts: { $ne: 1 } } },
+      { $group: { _id: '$attempts', count: { $sum: 1} } },
+    ]).toArray()
+
+    if (isNil(cursorArray) || cursorArray.length === 0) return []
+
+    return cursorArray.map( (item: AnchorRetryDAOResult) => ({ attempts: item._id, count: item.count }))
```
Comment (kennylavender) [inline src/API/IPFSDirectoryHashDAO.ts]: Please break this line into multiples lines.

File: src/API/IPFSDirectoryHashDAO.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    this.collection = collection
+  }
+
+  readonly getAnchorRetryHealth: getAnchorRetryHealth = async () => {
+    const cursorArray = await this.collection.aggregate([
+      { $match: { attempts: { $ne: 1 } } },
+      { $group: { _id: '$attempts', count: { $sum: 1} } },
+    ]).toArray()
+
+    if (isNil(cursorArray) || cursorArray.length === 0) return []
```
Comment (kennylavender) [inline src/API/IPFSDirectoryHashDAO.ts]: is ` || cursorArray.length === 0` this part needed? It should already be an empty array I would think?

File: src/API/IPFSDirectoryHashDAO.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { AnchorRetryDAOResult, getAnchorRetryHealth } from 'Interfaces'
+import { Collection } from 'mongodb'
+import { isNil } from 'ramda'
+
+export interface Entry {
+  hash: string
```
Comment (lautarodragan) [inline src/API/IPFSDirectoryHashDAO.ts]: readonly?

**Comment by krobi64** [comment]
Closing in favor of a slightly different approach

**Pull request closed without merging**